### PR TITLE
F/new channel methods

### PR DESCRIPTION
**Pull request by AshleyPoole**
This PR adds the ability to:
- Create channel
- Join channel
- Set channel purpose
- Set channel topic
- Archive channel

All new methods have accompanying tests using the format setup already in this repo.

This PR implements the feature request in issue (feature request) #11.

### CODE REVIEW

**Comment by AshleyPoole** [comment]
@Workshop2 ready for your review.

File: src/SlackConnector/ISlackConnection.cs [hunk 1]
```diff
@@ -86,12 +86,37 @@ public interface ISlackConnection
...
         Task<IEnumerable<SlackUser>> GetUsers();
 
-            /// <summary>
+        /// <summary>
         /// Opens a DM channel to a user. Required to PM someone.
         /// </summary>
         Task<SlackChatHub> JoinDirectMessageChannel(string user);
 
         /// <summary>
+        /// Creates a channel.
```
Comment (Workshop2) [inline src/SlackConnector/ISlackConnection.cs]: Wrong XML comment

File: src/SlackConnector/ISlackConnection.cs [hunk 2]
```diff
@@ -86,12 +86,37 @@ public interface ISlackConnection
...
+        /// </summary>
+        Task<SlackChatHub> JoinChannel(string channelName);
+
+        /// <summary>
+        /// Joins a channel.
```
Comment (Workshop2) [inline src/SlackConnector/ISlackConnection.cs]: And again

File: tests/SlackConnector.Tests.Unit/Connections/Clients/Flurl/FlurlChannelClientTests.cs [hunk 3]
```diff
@@ -64,6 +65,166 @@ public void Dispose()
...
+            var expectedResponse = new StandardResponse()
+            {
+                Ok = true
+            };
+
+            _httpTest.RespondWithJson(expectedResponse);
+
+            // when
+            var result = true;
+            try
```
Comment (Workshop2) [inline tests/SlackConnector.Tests.Unit/Connections/Clients/Flurl/FlurlChannelClientTests.cs]: Should let exceptions bubble up and be caught by the test harness

**Review by Workshop2 — CHANGES_REQUESTED** [review]
Wow, amazing work. A couple of bits of feedback.

**Comment by AshleyPoole** [comment]
@Workshop2 corrections made.

**Comment by Workshop2** [comment]
Thanks @ashleypoole

Fantastic work

**Comment by AshleyPoole** [comment]
Are you able to publish the nuget package for the latest version @Workshop2? I've checked nuget just but it's showing the old previous version.

**Comment by Workshop2** [comment]
Done

**Pull request merged**